tests/tracking/test_tracking.py

```diff
@@ -120,9 +117,8 @@ def test_set_experiment_with_zero_id(reset_mock, reset_active_experiment):
 
 
 def test_start_run_context_manager(tracking_uri_mock):
-    first_run = start_run()
-    first_uuid = first_run.info.run_uuid
-    with first_run:
+    with start_run() as first_run:
```

Comment:
I made a number of changes like this that try to wrap the return value of `start_run` in a with block immediately. Otherwise, if intermediate logic between the `start_run` and `with` block throws, almost every other test in this suite fails because the run created by `start_run` is never terminated & MLflow complains that that run is already active. This was sufficiently annoying while debugging that I made the change in this PR, but I could defer it.